File: make_sql.js
const fs = require('fs');

const raw = fs.readFileSync('supabase/demo_data.csv', 'utf16le');
const chunks = raw.split(/\r?\n\r?\n/).map(c => c.trim()).filter(c => c);

const tables = ['demo_area', 'demo_role', 'demo_user', 'demo_kpi', 'demo_target', 'demo_data'];
let sql = '-- SCRIPT PARA SUBIR TABLAS A LA DEMO REAL\n\n';

for (let i = 0; i < tables.length; i++) {
    const tName = tables[i];
    if (!chunks[i]) break;
    const lines = chunks[i].split(/\r?\n/).map(l => l.trim()).filter(l => l);
    const header = lines[0].split(',').map(h => h.trim());

    sql += 'CREATE TABLE IF NOT EXISTS ' + tName + ' (\n';
    const defs = header.map((col, idx) => {
        let type = 'VARCHAR(255)';
        if (col.includes('id') && col !== 'center_id') type = 'INTEGER';
        else if (col.includes('value')) type = 'DECIMAL(10,2)';
        else if (col.includes('description') || col.includes('method') || col.includes('permissions')) type = 'TEXT';

        return '    ' + col + ' ' + type + (idx === 0 ? ' PRIMARY KEY' : '');
    });
    sql += defs.join(',\n') + '\n);\n';
    sql += 'TRUNCATE TABLE ' + tName + ' CASCADE;\n\n';

    if (lines.length > 1) {
        sql += 'INSERT INTO ' + tName + ' (' + header.join(', ') + ') VALUES\n';
        const values = lines.slice(1).map(l => {
            let parts = l.split(',');
            // If a row has commas inside values, this simple split would break, 
            // but the data provided by user has no commas within values.
            return '(' + parts.map((p, j) => {
                p = p.trim().replace(/'/g, "''");
                const col = header[j];
                if ((col.includes('id') && col !== 'center_id') || col.includes('value')) {
                    return p; // numeric
                }
                return "'" + p + "'"; // string
            }).join(', ') + ')';
        });
        sql += values.join(',\n') + ';\n\n';
    }
}

fs.writeFileSync('supabase/demo_real_nexus.sql', sql, 'utf8');
console.log('Done generating supabase/demo_real_nexus.sql');

File: make_sql.py
import csv
import re

sections = []
current_section = []
with open('supabase/demo_data.csv', 'r') as f:
    for line in f:
        line = line.strip()
        if not line:
            if current_section:
                sections.append(current_section)
                current_section = []
        else:
            current_section.append(line)
if current_section:
    sections.append(current_section)

table_names = ['area', 'role', 'users', 'kpi', 'target', 'data']

with open('supabase/demo_real_nexus.sql', 'w') as out:
    out.write("-- ESTE SCRIPT CREA LAS TABLAS Y LOS DATOS PARA LA DEMO REAL\n\n")
    
    for i, section in enumerate(sections):
        if i >= len(table_names): break
        
        table_name = "demo_" + table_names[i]
        header = section[0].split(',')
        columns = [h.strip() for h in header]
        
        if table_name == 'demo_users':
           out.write(f"CREATE TABLE IF NOT EXISTS {{table_name}} (\n")
        else:
           out.write(f"CREATE TABLE IF NOT EXISTS {{table_name}} (\n")
           
        # Just use reasonable types
        # Very simple schema inference
        cols_sql = []
        for col in columns:
           if 'id' in col and not col == 'center_id':
              cols_sql.append(f"  {{col}} INTEGER")
           elif 'value' in col:
              cols_sql.append(f"  {{col}} DECIMAL(10,2)")
           else:
              cols_sql.append(f"  {{col}} VARCHAR(255)")
              
        cols_sql[0] += " PRIMARY KEY" if not table_name == 'demo_data' else ""
        out.write(",\n".join(cols_sql))
        out.write("\n);\n")
        
        out.write(f"TRUNCATE TABLE {{table_name}} CASCADE;\n\n")
        
        out.write(f"INSERT INTO {{table_name}} ({{', '.join(columns)}}) VALUES\n")
        
        values_lines = []
        for row in section[1:]:
            parts = [p.strip() for p in row.split(',')]
            val_parts = []
            for j, p in enumerate(parts):
                if 'id' in columns[j] and columns[j] != 'center_id':
                   val_parts.append(p)
                elif 'value' in columns[j]:
                   val_parts.append(p)
                else:
                   safe_p = p.replace("'", "''")
                   val_parts.append(f"'{safe_p}'")
            values_lines.append("(" + ", ".join(val_parts) + ")")
            
        out.write(",\n".join(values_lines))
        out.write(";\n\n")

print("SQL file generated successfully.")
